### PR TITLE
LPD-24312 Add necessary tests

### PR DESCRIPTION
**Pull request by BarbaraCabrera**
## Motivation

[Link to ticket](https://liferay.atlassian.net/browse/LPD-24312)

image

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+		Assert.assertNotNull(
+			_assetListEntryLocalService.fetchAssetListEntry(
+				assetListEntry.getExternalReferenceCode(),
+				assetListEntry.getGroupId()));
+
+		Assert.assertEquals(
+			"externalReferenceCode", assetListEntry.getExternalReferenceCode());
+	}
+
```
Comment (ealonso) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java:62]: @BarbaraCabrera Could you please add a new test to check Duplicate ExternalReferenceCode

**Comment by BarbaraCabrera** [comment]
ci:test:sf

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	@Test(
+		expected = DuplicateAssetListEntryExternalReferenceCodeException.class
+	)
+	public void testAddAssetListEntryWithExistingExternalReferenceCode()
+		throws PortalException {
+
+		_addAssetListEntry("externalReferenceCode", "Asset List Title");
```
Comment (ealonso) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java:70]: @BarbaraCabrera Randomize the names for AssetList in all file and remove the empty line

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+				assetListEntry.getGroupId()));
+	}
+
+	private AssetListEntry _addAssetListEntry(
+			String externalReferenceCode, String title)
+		throws PortalException {
+
+		return _assetListEntryLocalService.addAssetListEntry(
+			externalReferenceCode, TestPropsValues.getUserId(),
+			_group.getGroupId(), title, 0,
```
Comment (ealonso) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java:99]: @BarbaraCabrera instead of using 0 please use Constants

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ */
+@RunWith(Arquillian.class)
+public class AssetListEntryLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
```
Comment (ealonso) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryLocalServiceTest.java:41]: @BarbaraCabrera PermissionCheckerMethodTestRule is it necessary?

**Comment by ealonso** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/15622
